main.tf

```
@@ -92,3 +100,57 @@ resource "kubernetes_namespace" "namespace" {
     labels = each.value.labels
   }
 }
+
+resource "kubernetes_network_policy" "isolated" {
+  for_each = {
+    for p in local.drupal_network_policy_list : p.namespace => p.project_name if p.network_policy == "isolated"
+  }
+
+  metadata {
+    name      = "network-policy-isolated-${each.value}"
+    namespace = each.key
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {}
+    }
+
+    policy_types = ["Ingress"]
+
+    ingress {
+      from {
+        pod_selector {}
+      }
+    }
+  }
+
+  depends_on = [
+    kubernetes_namespace.namespace
+  ]
+}
+
+resource "kubernetes_network_policy" "restricted" {
```

```suggestion
resource "kubernetes_network_policy_v1" "restricted" {
```

Use [`kubernetes_network_policy_v1`](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/network_policy_v1) resource type.